fluss-common/src/main/java/org/apache/fluss/row/arrow/vectors/ArrowMapColumnVector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.fluss.row.arrow.vectors;
+
+import org.apache.fluss.row.InternalMap;
+import org.apache.fluss.row.columnar.ColumnVector;
+import org.apache.fluss.row.columnar.ColumnarMap;
+import org.apache.fluss.row.columnar.MapColumnVector;
+import org.apache.fluss.shaded.arrow.org.apache.arrow.vector.FieldVector;
+import org.apache.fluss.shaded.arrow.org.apache.arrow.vector.complex.MapVector;
+import org.apache.fluss.shaded.arrow.org.apache.arrow.vector.complex.StructVector;
+import org.apache.fluss.types.DataType;
+import org.apache.fluss.utils.ArrowUtils;
+
+/** ArrowMapColumnVector is a wrapper class for Arrow MapVector. */
+public class ArrowMapColumnVector implements MapColumnVector {
+    private boolean inited = false;
+    private final FieldVector vector;
+    private final DataType keyType;
+    private final DataType valueType;
+    private ColumnVector keyColumnVector;
+    private ColumnVector valueColumnVector;
+
+    public ArrowMapColumnVector(FieldVector vector, DataType keyType, DataType valueType) {
+        this.vector = vector;
+        this.keyType = keyType;
+        this.valueType = valueType;
+    }
+
+    private void init() {
+        if (!inited) {
+            FieldVector mapVector = ((MapVector) vector).getDataVector();
+            StructVector structVector = (StructVector) mapVector;
+            FieldVector keyVector = structVector.getChildrenFromFields().get(0);
+            FieldVector valueVector = structVector.getChildrenFromFields().get(1);
```

Comment:
Instead of direct access, you should use  
```java
structVector.getChild(MapVector.KEY_NAME)
```  
and  
```java
structVector.getChild(MapVector.VALUE_NAME)
```  
to obtain the `keyVector` and `valueVector`, respectively.

Additionally, to maintain consistency with `ArrowArrayColumnVector` and `ArrowRowColumnVector`, it’s better to move the initialization of `keyColumnVector` and `valueColumnVector` **outside** of `ArrowMapColumnVector`.  

This also allows us to avoid exposing `ArrowUtils.createArrowColumnVector` as `public`, preserving better encapsulation and alignment with existing patterns.